CesiumVectorData/include/CesiumVectorData/VectorRasterizer.h

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include "VectorStyle.h"
+
+#include <CesiumGeometry/Rectangle.h>
+#include <CesiumGeospatial/CartographicPolygon.h>
+#include <CesiumGeospatial/Ellipsoid.h>
+#include <CesiumGeospatial/GlobeRectangle.h>
+#include <CesiumGltf/ImageAsset.h>
+#include <CesiumUtility/Color.h>
+#include <CesiumUtility/IntrusivePointer.h>
+#include <CesiumUtility/ReferenceCounted.h>
+#include <CesiumVectorData/GeoJsonObject.h>
+
+#include <blend2d.h>
+#include <blend2d/context.h>
+#include <blend2d/image.h>
+
+#include <span>
+
+namespace CesiumVectorData {
+
+/**
+ * @brief Rasterizes vector primitives into a \ref CesiumGltf::ImageAsset.
+ */
+class VectorRasterizer {
+public:
+  /**
+   * @brief Creates a new \ref VectorRasterizer representing the given rectangle
+   * on the globe.
+   *
+   * @param bounds The bounds on the globe that this rasterizer's canvas will
+   * cover.
+   * @param imageAsset The destination image asset. This \ref
+   * CesiumGltf::ImageAsset must be four channels, with
+   * only one byte per channel (RGBA32).
+   * @param mipLevel The mip level that the rasterizer should rasterize for the
+   * image.
+   * @param ellipsoid The ellipsoid to use.
+   */
+  VectorRasterizer(
+      const CesiumGeospatial::GlobeRectangle& bounds,
+      CesiumUtility::IntrusivePointer<CesiumGltf::ImageAsset>& imageAsset,
+      uint32_t mipLevel = 0,
+      const CesiumGeospatial::Ellipsoid& ellipsoid =
+          CesiumGeospatial::Ellipsoid::WGS84);
+
+  /**
+   * @brief Draws a \ref CesiumGeospatial::CartographicPolygon to the canvas.
+   *
+   * @param polygon The polygon to draw.
+   * @param style The \ref VectorStyle to use when drawing the polygon.
+   */
+  void drawPolygon(
+      const CesiumGeospatial::CartographicPolygon& polygon,
+      const VectorStyle& style);
+
+  /**
+   * @brief Draws a set of linear rings representing a polygon and its holes to
+   * the canvas.
+   *
+   * @param polygon The polygon to draw. It is assumed to have right-hand
+   * winding order (exterior rings are counterclockwise, holes are clockwise) as
+   * is the case in GeoJSON. The coordinates should be specified in degrees.
+   * @param style The \ref VectorStyle to use when drawing the polygon.
+   */
+  void drawPolygon(
+      const std::vector<std::vector<glm::dvec3>>& polygon,
+      const VectorStyle& style);
+
+  /**
+   * @brief Draws a polyline (a set of multiple line segments) to the canvas.
+   *
+   * @param points The set of points making up the polyline. The coordinates
+   * should be specified in degrees.
+   * @param style The \ref VectorStyle to use when drawing the polyline.
+   */
+  void drawPolyline(
+      const std::span<const glm::dvec3>& points,
+      const VectorStyle& style);
+
+  /**
+   * @brief Rasterizes the provided `GeoJsonObject` to the canvas.
+   *
+   * Polygons are equivalent to calls to `drawPolygon`. Polylines are equivalent
+   * to calls to `drawPolyline`. Points are currently not drawn.
```

Comment:
I guess I'd expect this method to draw everything in the GeoJsonObject. For example, if it were a `FeatureCollection`, it would draw every line and polygon in any feature in the collection. Is that a crazy expectation? If so, we should at least update the docs to make it very clear that this is not the case. It mentions that points are not drawn, but it doesn't mention other things.